modules/nf-core/fastqc/main.nf

```diff
@@ -20,6 +20,10 @@ process FASTQC {
 
     script:
     def args = task.ext.args ?: ''
+    def cpu = task.cpus
+    def memory = task.memory.toMega() > 10000 ? 10000 : task.memory.toMega()
+    if (memory/250 < cpu)
+        cpu = Math.floor(memory/250).toInteger()
```

Comment:
I don't think you should modify the number of cpu's inside the module 😨

Reply:
Because of this (from the manual):

>     -t --threads    Specifies the number of files which can be processed
>                 simultaneously.  Each thread will be allocated 250MB of
>                 memory so you shouldn't run more threads than your
>                 available memory will cope with, and not more than
>                 6 threads on a 32 bit machine

So I thought a thread must have at least 250MB memory, so if one gives 500Mb and 7 threads, there won't be enough memory to all the threads